packages/angular/projects/clr-angular/src/data/datagrid/datagrid-action-item.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Directive, ElementRef } from '@angular/core';
+
+/**
+ * @remark
+ * This directive is used to identify action buttons within the single row action menu and to give
+ * them the 'menuitem' aria role.
+ */
+@Directive({
+  selector: '.action-item',
```

Comment:
I think we need to be careful with this. Even though we only use `action-item` in the main-renderer.spec.ts on line 334 for the Clarity codebase, this could cause consumers problems if they also have elements with `class="menu-item" on anything. If we go this route we might have to prefix it to ty and create a namespace around it and then change the classes inside the datagrid.

Reply:
Good call Matt. Unfortunately Angular doesn't allow you to use any CSS selector, so it would not be possible to make this directive work better without targetting potentially any `class="action-item"`.

I think a better solution here is to make this an attribute directive, which would use host bindings to set the class and the role instead of trying to reach in and use the native Element below (which would break SSR currently).

This is untested, but basically this would be the class:

```
@Directive({
  selector: '[clrDgActionItem]',
  host: {
    '[class.action-item]': true,
    role: 'menuitem',
  }
})
export class ClrDatagridActionItem {}
```

Then the API would be 

```
<clr-dg-action-overflow>
        <button clrDgActionItem (click)="onEdit(user)">Edit</button>
        <button clrDgActionItem (click)="onDelete(user)">Delete</button>
</clr-dg-action-overflow>
```

This is not breaking the existing API, but for anyone who wants this a11y fix they'd update their usage.

Reply:
I was also thinking about the generic class names but unfortunately it's hard to change them at this point.  I'm also worried about relying on users to do anything to fix a11y issues.  

What about having `clr-dg-action-item` dynamically add the menuitem role to all buttons within it using `@ViewChildren`?  If that seems too invasive I can take the approach above.

Reply:
Frankly, this bug is easily resolved by apps who can just add the right role to their buttons. Anything we do is a bonus.

This approach might also work, but needs to be careful of dynamic rendering scenarios. It must watch for changes to the list. Also, the direct nativeElement access below will break in server side rendering, so we need to use the Angular renderer.

I worry that this might have a problem or two around dynamic items because of lifecycles. We see this in a few places, like list of tabs are rendered too late and not picked up by the screen reader. If that is the case, then it would be better for apps to directly apply the roles since that would render the menu items in regular time vs in a potential later cycle (due to renderer).

Reply:
All good points, updated with your suggestion above.